Jellyfin Server/AppDelegate.swift

```diff
@@ -74,7 +43,13 @@ class AppDelegate: NSObject, NSApplicationDelegate {
         }
 
         jellyfinProcess.launchPath = jellyfinPath
-        jellyfinProcess.arguments = ["--webdir", webUIPath, "--ffmpeg", ffmpegPath, "--datadir", applicationSupportJellyfinFolder.path]
+        jellyfinProcess.arguments = ["--webdir", webUIPath, "--ffmpeg", ffmpegPath, "--datadir"]
+        
+        if directoryExists(path: localShareJellyfinFolder.path) {
+            jellyfinProcess.arguments?.append(localShareJellyfinFolder.path)
+        } else {
+            jellyfinProcess.arguments?.append(applicationSupportJellyfinFolder.path)
```

Comment:
If you said that _the_ problem is that this folder never existed in the first place, then no files were properly moved to it. I also don't care as much as I did back then to move things to this folder, so we can just ignore it and don't need this conditional.

This would only apply to users from the issue that have manually migrated things over.

Reply:
No, if we don't care about this the new users installed this from 10.9 fresh will have their instances stopped working. The 10.9 creates this folder in Application Support by default so we really need to check if we want to sue the old path or not.

Reply:
Sorry, you're right for fresh users. However I think that this split is only going to cause issues for us in the future and think that a message of "if you installed 10.9 fresh, please delete and re-install" is warranted.

Reply:
This is the same as "we already fucked you once by forcing you start clean, now we want to fuck you once more".

I don't see any problems for having multiple data dir. On other platforms the user have even more freedom on where the data is located and has not caused problems yet.

Reply:
No need to get explicit or rude. I don't have a problem admitting we broke something. If this does ever become an issue in the future, then idc to address it.

Reply:
We had this problem once upon a time in the Windows release. If you installed with a script in the early days, your folders were in one location (what the OS provided to XDG defaults), and the installer was built to conform to Windows "proper" folders (which were different from the XDG defaults). Sadly I just had to write a detection for it in the installer: https://github.com/jellyfin/jellyfin-server-windows/blob/299639e13ea79758f61d47799881810d407090cd/nsis/jellyfin.nsi#L530-L538

Reply:
Then let's do the same for macOS now. I really don't think force the user to start clean is anything close to user friendly, especially for a bugfix update.